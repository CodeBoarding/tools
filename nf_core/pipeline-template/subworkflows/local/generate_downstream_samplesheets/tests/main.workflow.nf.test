nextflow_workflow {

    name "Test Workflow GENERATE_DOWNSTREAM_SAMPLESHEETS"
    script "../main.nf"
    workflow "GENERATE_DOWNSTREAM_SAMPLESHEETS"
    tag 'subworkflows'
    tag 'generate_downstream_samplesheets'
    tag 'subworkflows/generated_downstream_samplesheets'

    test("Test worfklow rnaseq") {
        when {
            params {
                outdir                                = "."
                generate_pipeline_samplesheets        = 'rnaseq'
                generate_pipeline_samplesheets_format = 'csv'
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [id: 'test-pe', single_end: false, reads: ['test_1.fastq.gz', 'test_2.fastq.gz']]
                    ],
                    [
                        [id: 'test-se', single_end: true, reads: ['test_1.fastq.gz']]
                    ]
                )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    [
                        "${params.outdir}/downstream_samplesheets/rnaseq.csv"
                    ]).match()
                }
            )
        }
    }

}
