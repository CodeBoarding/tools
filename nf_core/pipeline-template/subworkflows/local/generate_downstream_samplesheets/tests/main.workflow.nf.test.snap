{
    "Test worfklow rnaseq": {
        "content": [
            [
                "./downstream_samplesheets/rnaseq.csv"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-30T09:34:46.912767154"
    }
}